pyiron/sphinx/base.py

```diff
@@ -614,22 +616,57 @@ def _set_kpoints(
             raise ValueError("manual_kpoints is not implemented in SPHInX yet")
         if weights is not None:
             raise ValueError("manual weights are not implmented in SPHInX yet")
-        if scheme != "MP":
-            raise ValueError("only Monkhorst-Pack mesh is implemented in SPHInX")
-        if mesh is not None:
-            self.input["KpointFolding"] = (
-                "[" + str(mesh[0]) + ", " + str(mesh[1]) + ", " + str(mesh[2]) + "]"
-            )
-        if center_shift is not None:
-            self.input["KpointCoords"] = (
-                "["
-                + str(center_shift[0])
-                + ", "
-                + str(center_shift[1])
-                + ", "
-                + str(center_shift[2])
-                + "]"
+        if scheme == "MP":
+            self._kpoints_odict = None
+            if mesh is not None:
+                self.input["KpointFolding"] = (
+                        "[" + str(mesh[0]) + ", " + str(mesh[1]) + ", " + str(mesh[2]) + "]"
+                )
+            if center_shift is not None:
+                self.input["KpointCoords"] = (
+                        "["
+                        + str(center_shift[0])
+                        + ", "
+                        + str(center_shift[1])
+                        + ", "
+                        + str(center_shift[2])
+                        + "]"
+                )
+        elif scheme == "Line":
+            if n_trace is None:
+                raise ValueError("n_trace has to be defined")
+            high_symmetry_points = self.structure.get_high_symmetry_points()
+            if high_symmetry_points is None:
+                raise ValueError("high_symmetry_points has to be defined")
```

Comment:
Is it up to the user to define `high_symmetry_points`?

Reply:
I thought they are given by `high_symmetry_points = self.structure.get_high_symmetry_points()` - but maybe the user should have the option to overwrite those.

Reply:
I'm fine with `high_symmetry_points = self.structure.get_high_symmetry_points()`, but then the error message is misleading.

Reply:
The user has to define the `high_symmetry_points` for the `structure` object. Since they are structural quantities, they should not be defined within `set_kpoints`.

Reply:
@samwaseda you're right. I'll change the message.